node/src/components/sync_leaper/leap_activity.rs

```diff
@@ -0,0 +1,482 @@
+                .cloned(),
+        );
+
+        let (actual_sync_leap, actual_peers) = leap_activity.best_response().unwrap();
+
+        assert_eq!(actual_peers.len(), 1);
+        assert_eq!(actual_peers.get(0).unwrap(), &peer_1.0);
+        assert_eq!(actual_sync_leap, sync_leap);
+    }
+
+    #[test]
+    fn best_response_with_multiple_peers() {
+        let mut rng = TestRng::new();
+
+        // Create 10 sync leaps, each with a distinct height. The height is not greater than 10
+        let sync_leap_identifier = SyncLeapIdentifier::sync_to_tip(BlockHash::random(&mut rng));
+        let mut heights: Vec<u64> = (0..10).collect();
+        heights.shuffle(&mut rng);
+        let mut peers_with_sync_leaps: HashMap<_, _> = heights
+            .iter()
+            .map(|height| {
+                let block = make_random_block_with_height(&mut rng, *height);
+                let sync_leap = make_sync_leap_with_trusted_block_header(block.header().clone());
+                (
+                    NodeId::random(&mut rng),
+                    PeerState::Fetched(Box::new(sync_leap)),
+                )
+            })
+            .collect();
+
+        // Add another peer with the best response.
+        let block = make_random_block_with_height(&mut rng, 500);
+        let best_sync_leap = make_sync_leap_with_trusted_block_header(block.header().clone());
+        let peer_1_best_node_id = NodeId::random(&mut rng);
+        peers_with_sync_leaps.insert(
+            peer_1_best_node_id,
+            PeerState::Fetched(Box::new(best_sync_leap.clone())),
+        );
+        let mut leap_activity = LeapActivity {
+            sync_leap_identifier,
+            peers: peers_with_sync_leaps.clone(),
+            leap_start: Instant::now(),
+        };
+
+        let (actual_sync_leap, actual_peers) = leap_activity.best_response().unwrap();
+
+        // Expect only a single peer with the best sync leap.
+        assert_eq!(actual_peers.len(), 1);
+        assert_eq!(actual_peers.get(0).unwrap(), &peer_1_best_node_id);
+        assert_eq!(actual_sync_leap, best_sync_leap);
+
+        // Add two more peers with even better response.
+        let block = make_random_block_with_height(&mut rng, 1000);
+        let best_sync_leap = make_sync_leap_with_trusted_block_header(block.header().clone());
+        let peer_2_best_node_id = NodeId::random(&mut rng);
+        let peer_3_best_node_id = NodeId::random(&mut rng);
+        leap_activity.peers.extend(
+            [
+                (
+                    peer_2_best_node_id,
+                    PeerState::Fetched(Box::new(best_sync_leap.clone())),
+                ),
+                (
+                    peer_3_best_node_id,
+                    PeerState::Fetched(Box::new(best_sync_leap.clone())),
+                ),
+            ]
+            .iter()
+            .cloned(),
+        );
+
+        let (actual_sync_leap, mut actual_peers) = leap_activity.best_response().unwrap();
+
```

Comment:
nit: At this point I'd add a peer with a worse response, just to make sure adding a worse valid response doesn't break things.

Reply:
Good idea - test has been extended in 2cecd103bad13e3d777c9fa249e38d2d4d766556